src/fsharp/NameResolution.fs

```diff
@@ -2331,7 +2334,21 @@ let rec private ResolveTypeLongIdentInModuleOrNamespace (ncenv:NameResolver) (ty
         let tcrefs = LookupTypeNameInEntityMaybeHaveArity (ncenv.amap, id.idRange, id.idText, typeNameResInfo.StaticArgsInfo, modref)
         match tcrefs with 
         | _ :: _ -> tcrefs |> CollectResults (fun tcref -> success(resInfo,tcref))
-        | [] -> raze (UndefinedName(depth,FSComp.SR.undefinedNameType,id,[]))
+        | [] ->
+            let possibleTypes =
+                modref.ModuleOrNamespaceType.AllEntities
+                |> Seq.map (fun e -> e.DisplayName)
+                |> Set.ofSeq
+
+            let details =
+                ErrorResolutionHints.CreatePredictionText
+                    FSComp.SR.undefinedNameRecordLabelDetails
+                    id.idText
+                    possibleTypes
+
+            let errorTextF s = FSComp.SR.undefinedNameTypeIn(s,modref.ToString())
```

Comment:
Using `ToString` is non-idiomatic in the compiler - ToString values are generally only used for debug displays - please use `DisplayText` or some other explicit formatting function.